taipy/core/data/couchbase_collection_datanode.py

```diff
@@ -0,0 +1,183 @@
+            properties = {}
+        required = self._REQUIRED_PROPERTIES
+        if missing := set(required) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties {', '.join(missing)} were not informed and are required."
+            )
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            owner_id,
+            parent_ids,
+            last_edit_date,
+            edits,
+            version or _VersionManagerFactory._build_manager()._get_latest_version(),
+            validity_period,
+            edit_in_progress,
+            editor_id,
+            editor_expiration_date,
+            **properties,
+        )
+
+
+         # Create a Couchbase connection using the provided properties.
+         # For more information on connecting to Couchbase, see:
+    # https://docs.couchbase.com/python-sdk/current/hello-world/start-using-sdk.html
+        try:
+            self.cluster = Cluster(
+                f"couchbase://{properties.get(self.__DB_HOST_KEY, self.__DB_HOST_DEFAULT)}",
+                ClusterOptions(
+                    PsswordAuthenticator(
+                        properties.get(self.__DB_USERNAME_KEY, ""),
+                        properties.get(self.__DB_PASSWORD_KEY, "")
+                    )
+                )
+            )
+            bucket = self.cluster.bucket(properties.get(self.__DB_NAME_KEY))
+            self.collection = bucket.collection(properties.get(self.__COLLECTION_KEY))
+
+        except CouchbaseException as e:
+            raise ConnectionError(f"Could not connect to Couchbase: {e}")
+
+        self._TAIPY_PROPERTIES.update(
+            {
+                self.__COLLECTION_KEY,
+                self.__DB_NAME_KEY,
+                self.__DB_USERNAME_KEY,
+                self.__DB_PASSWORD_KEY,
+                self.__DB_HOST_KEY,
+                self.__DB_PORT_KEY,
+            }
+        )
+
+    @classmethod
+    def storage_type(cls) -> str:
+        """Return the storage type of the data node: "couchbase_collection"."""
+        return cls.__STORAGE_TYPE
+
+    def _read(self):
+        """Read all documents from the Couchbase collection."""
+        try:
+            query = f"SELECT * FROM `{self.collection.name}`"
+            result = self.cluster.query(query)
+
+            return [doc.content_as[dict] for doc in documents]
+        except Exception as e:
+             print(f"An error occurred: {e}")
+             return []
+
+    def _write(self,data : Union[Dict, List[Dict]]):
+           """Write Documents to the Couchbase collection."""
+        try:
+            if isinstance(data, dict):
+                self.collection.upsert(data['id'],data)
+            elif isinstance(data, list):
+                for item in data:
+                   self.collection.upsert(item['id'], item)
+       except CouchbaseException as e:
+            print(f"An error occurred while writing:{e}")
```

Comment:
```suggestion
        try:
            if isinstance(data, dict):
                self.collection.upsert(data['id'],data)
            elif isinstance(data, list):
                for item in data:
                   self.collection.upsert(item['id'], item)
        except CouchbaseException as e:
            print(f"An error occurred while writing:{e}")
```

The `except` is misalign